packages/compiler/server/serverlib.ts

```diff
@@ -618,9 +623,53 @@ export function createServer(host: ServerHost): Server {
   async function addImportCompletion(
     program: Program,
     document: TextDocument,
-    completions: CompletionList
+    completions: CompletionList,
+    node: StringLiteralNode
   ) {
-    await addLibraryImportCompletion(program, document, completions);
+    if (node.value.startsWith("./") || node.value.startsWith("../")) {
+      await addRelativePathCompletion(program, document, completions, node);
+    } else if (!node.value.startsWith(".")) {
+      await addLibraryImportCompletion(program, document, completions);
+    }
+  }
+
+  async function addRelativePathCompletion(
+    program: Program,
+    document: TextDocument,
+    completions: CompletionList,
+    node: StringLiteralNode
+  ) {
+    const documentPath = getPath(document);
+    const documentFile = getBaseFileName(documentPath);
+    const documentDir = getDirectoryPath(documentPath);
+    const nodevalueDir = hasTrailingDirectorySeparator(node.value)
+      ? node.value
+      : getDirectoryPath(node.value);
+    const maincadl = resolvePath(documentDir, nodevalueDir);
+    const listFiles = (await program.host.readDir(maincadl)).filter(
```

Comment:
nit: `files` is a better name than `listFiles` which sounds like it would hold a function.